### PR TITLE
feat(web3hub): main and search screen first iteration [LIVE-13176]

### PR DESCRIPTION
**Pull request by Justkant**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - none feature behind FF

### 📝 Description

Prepared a good template for the other screens
Fixed a lot of small details with the headers and lists while working on it
Implemented some integration tests for the current screens and logic
Only mocks to fetch for now
Started i18n

Also for the changes on the native-ui lib you can see the warning that was printed in the screenshot below, I'm not sure if there is an impact to my change and I suppose that I could also just remove the line

Screenshot 2024-07-10 at 16 08 42

### ❓ Context

- **JIRA or GitHub link**: [LIVE-13176] [LIVE-13182] [LIVE-13183] [LIVE-13186]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-13176]: https://ledgerhq.atlassian.net/browse/LIVE-13176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mcayuelas-ledger — APPROVED** [review]

**Review by KVNLS — APPROVED** [review]
LGTM!

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/CategoriesList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React, { useMemo } from "react";
+import { FlashList } from "@shopify/flash-list";
+import { useQuery } from "@tanstack/react-query";
+import { Box } from "@ledgerhq/native-ui";
+import Badge from "./Badge";
+
+const categories = [
```
Comment (ComradeAERGO) [inline apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/CategoriesList/index.tsx:7]: [No Action] Would there be a way in the future to get those categories dynamically from DappRadar? (Maybe not) - Would be nice not to have to handle that manually.

Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/CategoriesList/index.tsx:7]: Everything in this PR is mocked data, the actual categories from the api are coming from DappRadar

**Review by ComradeAERGO — APPROVED** [review]
Great job !

**Pull request merged**